src/server/worker/master.go

```diff
@@ -917,6 +934,27 @@ func (a *APIServer) scaleUpWorkers() error {
 	return err
 }
 
+type cachedTags struct {
+	tag      *pfs.Tag
+	statsTag *pfs.Tag
+}
```

Comment:
This is kinda inefficient. `tag.Name` is exactly the same value as `hash` and `statsTag` is just that value with `_stats` appended to it. So We're storing 3 copies of the same value in the cache. Given that it's sized for 1000000 datums I think it's pretty worth it to cut down the size of the objects we're storing as much as possible. Actually while you're at it maybe we should take the `tag` and `statsTag` out of `ProcessResponse`. Since that's extra bandwidth / latency and we'll already be computing that on the master.